nxdrive/gui/view.py

```diff
@@ -296,13 +299,32 @@ def roleNames(self) -> Dict[int, bytes]:
     def set_items(
         self, items: List[Dict[str, Any]], parent: QModelIndex = QModelIndex()
     ) -> None:
-        self.beginRemoveRows(parent, 0, self.rowCount() - 1)
-        self.items.clear()
-        self.endRemoveRows()
+        if items and not self.shadow_item:
+            # Copy the first element from items and use it as shadow_item
+            self.shadow_item = items[0].copy()
+            self.shadow_item["shadow"] = True
 
-        self.beginInsertRows(parent, 0, len(items) - 1)
-        self.items.extend(items)
-        self.endInsertRows()
+        # Create the items list with real datas from items.
+        for n_item in items:
+            self.add_item(parent, n_item)
+
+        # Add shadow_items to complete items list up to limit
+        while len(self.items) < 5:  # Limit the items list size to 5
```

Comment:
That hardcoded limit is not an issue as it will always return 5 items maximum: the same number is hardcoded into: https://github.com/nuxeo/nuxeo-drive/blob/068e685ee796a47aec5484c36cd2f4b1824f8a01/nxdrive/gui/api.py#L211-L213

I think it is time to move that number to a constant:
```python
# Number of files displayed in the Direct Transfer window, onto the Moniroting tab
DT_MONITORING_SHOW_N_FILES = 5
```

And use it everywhere.

Reply:
BTW if you have a better constant name, go ahead :)